psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseValidation.java

```diff
@@ -0,0 +1,513 @@
+        }
+    }
+
+    private static String multipleSwitchedShuntString(String key, List<String> value, PsseVersion psseVersion) {
+        if (psseVersion.major() == V35) {
+            return String.format("SwitchedShunt: Multiple fixed shunts (%d) at bus %d with the same Id %s", value.size(), Integer.valueOf(key), value.get(0));
+        } else {
+            return String.format("SwitchedShunt: Multiple fixed shunts (%d) at bus %d", value.size(), Integer.valueOf(key));
+        }
+    }
+
+    private static PsseBus getRegulatingBus(List<PsseBus> psseBuses, Map<Integer, List<Integer>> buses, int ireg, int i) {
+        int regulatingId = i;
+        if (ireg != 0) {
+            regulatingId = ireg;
+        }
+
+        if (buses.containsKey(regulatingId)) {
+            return psseBuses.get(buses.get(regulatingId).get(0));
+        }
+
+        return null;
+    }
+
+    private static Map<String, List<String>> getDuplicates(Map<String, List<String>> busesMap) {
+        Map<String, List<String>> duplicatedBusMap = new HashMap<>();
+
+        busesMap.forEach((key, value) -> value.stream().collect(Collectors.groupingBy(s -> s))
+            .entrySet()
+            .stream()
+            .filter(e -> e.getValue().size() > 1)
+            .forEach(e -> duplicatedBusMap.put(key, e.getValue())));
+
+        return duplicatedBusMap;
+    }
+
+    private static void addBusesMap(Map<String, List<String>> busesMap, int busI, String id) {
+        String busString = String.format("%06d", busI);
+        busesMap.computeIfAbsent(busString, k -> new ArrayList<>()).add(id);
+    }
+
+    private static void addBusesMap(Map<String, List<String>> busesMap, int busI, int busJ, String ckt) {
+        String busString;
+        if (busI < busJ) {
+            busString = String.format("%06d-%06d", busI, busJ);
+        } else {
+            busString = String.format("%06d-%06d", busJ, busI);
+        }
+
+        busesMap.computeIfAbsent(busString, k -> new ArrayList<>()).add(ckt);
+    }
+
+    private static void addBusesMap(Map<String, List<String>> busesMap, int busI, int busJ, int busK, String ckt) {
+        String busString;
+        List<Integer> buses = new ArrayList<>();
+        buses.add(busI);
+        buses.add(busJ);
+        buses.add(busK);
+        Collections.sort(buses);
+        busString = String.format("%06d-%06d-%06d", buses.get(0), buses.get(1), buses.get(2));
+
+        busesMap.computeIfAbsent(busString, k -> new ArrayList<>()).add(ckt);
+    }
+
+    private static int firstBus(String busKey) {
+        String[] tokens = busKey.split("-");
+        return Integer.valueOf(tokens[0]);
+    }
+
+    private static int secondBus(String busKey) {
+        String[] tokens = busKey.split("-");
+        return Integer.valueOf(tokens[1]);
+    }
+
+    private static int thirdBus(String busKey) {
+        String[] tokens = busKey.split("-");
+        return Integer.valueOf(tokens[2]);
+    }
+
+    private List<String> warnings;
```

Comment:
list of warnings can be `final` (is created in the constructor)

Reply:
Done